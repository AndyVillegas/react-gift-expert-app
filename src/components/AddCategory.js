import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }
    const handleSubmit = (e)=>{
        e.preventDefault();
        if (valueIsValid()) {
            setCategories(categories=>[ inputValue,...categories]);
            clearInput();
        }
    }
    const clearInput = ()=>{
        setInputValue('');
    }
    const valueIsValid = ()=>inputValue.trim().length>2;
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}