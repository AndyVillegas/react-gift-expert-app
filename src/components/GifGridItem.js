import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
    return (
        <div className="card">
            <img src={url} alt={title?.length>0?title:'h'}/>
            <div>{title?.length>0?title:'No title'}</div>
        </div>
    )
}
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}