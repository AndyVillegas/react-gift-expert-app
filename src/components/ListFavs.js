import React from 'react'

export const ListFavs = () => {
    return (
        <div>
            
        </div>
    )
}
