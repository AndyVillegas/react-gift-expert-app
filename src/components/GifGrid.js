import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h3>{category}</h3>
        {loading && <div className="loading">Cargando...</div>}
        <div className="card-container animate__animated animate__fadeInLeft">
            {
                images.map(image => (
                    <GifGridItem
                        key={image.id}
                        {...image}
                    />
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}