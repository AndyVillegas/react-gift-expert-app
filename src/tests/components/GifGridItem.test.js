import { GifGridItem } from './../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en GifGridItem', () => {
    const title = 'Un título';
    const url = 'https://media2.giphy.com/media/AYy9YYXHmsGek/giphy.gif?cid=1a00d25bln92geglm8h1fk0vxrfixyibp3fzn9lye38wnwny&rid=giphy.gif';
    const wrapper = shallow(
        <GifGridItem title={title} url={url} />
    );

    test('debe de hacer match con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un div con el title', () => {
        const textP = wrapper.find('div>div').text();
        expect(textP).toBe(title);
    })

    test('debe de tener el alt igual al título y la url de las props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener la clase card', () => {
        const div = wrapper.find('div').at(0);
        expect(div.hasClass('card')).toBe(true);
        expect(div.prop('className')).toContain('card');
    });
})
