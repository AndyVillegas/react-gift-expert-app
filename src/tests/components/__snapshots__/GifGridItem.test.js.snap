// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de hacer match con el snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="Un título"
    src="https://media2.giphy.com/media/AYy9YYXHmsGek/giphy.gif?cid=1a00d25bln92geglm8h1fk0vxrfixyibp3fzn9lye38wnwny&rid=giphy.gif"
  />
  <div>
    Un título
  </div>
</div>
`;
